account_customer_wallet/tests/test_balance.py

```diff
@@ -68,3 +69,19 @@ def test_no_wallet_account(self):
 
         self.assertFalse(self.company_id.is_enabled_customer_wallet)
         self.assertEqual(self.partner.customer_wallet_balance, 0)
+
+    def test_payment(self):
+        """Prevent wallet to be negative, by blocking payments"""
+        self._create_move(credit=100)
+
+        # Sale and debit wallet (40)
+        invoice_1 = self._create_sale_invoice("out_invoice", 40)
+        # Try to debit wallet (New balance will be 60 should be OK)
+        self._create_payment(invoice_1, amount=40)
+        self.assertEqual(self.partner.customer_wallet_balance, 60)
```

Comment:
After creating `invoice_1`, assert that balance is (still) 100 as sanity check.

Reply:
Done.


(mark as solved)